chainer/backends/cuda.py

```diff
@@ -219,7 +218,8 @@ def get_device(*args):
 
 def _get_device(*args):
     for arg in args:
-        if type(arg) in _integer_types:
+        if isinstance(arg, _integer_types) and not numpy.issubdtype(
+                type(arg), numpy.bool_):
```

Comment:
Ah, I meant `bool`, not `numpy.bool_`.
(`isinstance(numpy.bool_, numpy.integer)` and `isinstance(numpy.bool_, six.integer_types)`) are both `False`, but `isinstance(bool, six.integer_types)` is `True`).

So the correct condition would be:
`if type(arg) is not bool and isinstance(arg, _integer_types):`

Reply:
Isn't my condition also correct? It works for both `numpy.bool`, `numpy.bool_` and `bool`.

Reply:
I see. So the outcome would be the same, but it does duplicate check and slower (`numpy.bool_` never satisfies the first condition). Also note that `numpy.bool` and `bool` are identical.

Reply:
Aha, bool and np.bool are identical. Then your solution is cleaner. Thanks! I will update it.